### ISSUE TITLE
Failed to Detect Deno

### ISSUE THREAD

**Issue by bfsmith** [denoland/vscode_deno#19]
I'm experiencing the same issue as https://github.com/justjavac/vscode-deno/issues/78; starting a new issue here since this seems to be the location of new development.

Mac Mojave 10.14.6
VS Code 1.45.0
VS Code Deno Extension version 1.27.0
Deno 1.0.0

# Environment
```
$ deno info
DENO_DIR location: "/Users/ME/.deno"
Remote modules cache: "/Users/ME/.deno/deps"
TypeScript compiler cache: "/Users/ME/.deno/gen"
$ deno eval 'console.log(JSON.stringify(Deno.version))'
{"deno":"1.0.0","v8":"8.4.300","typescript":"3.9.2"}
$ env | grep -i deno
DENO_INSTALL=/Users/ME/.deno
DENO_DIR=/Users/ME/.deno
$ which deno
/Users/ME/.deno/bin/deno
```

I use zsh as my main shell environment.

I tracked the error down to https://github.com/denoland/vscode_deno/blob/9a684b5a735b17b6bedae9a64e68f0223cfe0988/client/src/utils.ts#L64 and saw it's using `execa`. From the [execa README](https://github.com/sindresorhus/execa#execafile-arguments-options).
> Unless the shell option is used, no shell interpreter (Bash, cmd.exe, etc.) is used, so shell features such as variables substitution (echo $PATH) are not allowed.

I export `DENO_INSTALL` and `DENO_DIR` in my .zshrc file and add it to the path. Perhaps `execa` isn't finding Deno because it's not in the standard PATH locations?

**Comment by justjavac** [comment]
I will rewrite the deno executable finder

**Comment by yc51** [comment]
@justjavac 老兄，windows也提示在$PATH找不到deno,你代码写的在ProgramFiles里面找，windows默认安装路径是%userprofile%\.deno\

**Comment by justjavac** [comment]
> @justjavac 老兄，windows也提示在$PATH找不到deno,你代码写的在ProgramFiles里面找，windows默认安装路径是%userprofile%\.deno\

Windows also prompts that deno cannot be found in `$PATH`.

**Comment by justjavac** [comment]
maybe [which](https://www.npmjs.com/package/which) is a good choice

**Comment by bfsmith** [comment]
If I get some time today, I'll try and get the extension running locally and test out `which`

**Comment by bfsmith** [comment]
I was able to fix this without changing the extension by fully restarting VS Code. I think what happened was I had an instance of VS Code open before I installed Deno and set up the PATH to include it. When I opened a new VS Code instance and installed this extension, the new instance didn't rerun my `.zshrc` file to pick up the new PATH. I'm guessing VS Code sources it once and uses it for all instances (maybe this is Mac specific behavior).

After fully quitting VS Code and opening it back up, the extension was able to find Deno in the path. Getting `Deno.` intellisense and all. Thanks for your work on this @justjavac!

tl;dr: Make sure deno is on your path and fully quit VS Code, then open it back up and see if it works.

**Comment by justjavac** [comment]
Other people have encountered the same problem, so I should fix this problem

**Comment by willconant** [comment]
I had the same problem while using the remote-ssh plugin to work on a remote server. In my case, I had connected to the remote before installing deno, so I had to restart the VS Code agent on the *remote* before the plugin could find it. Restarting my local VS Code had no effect because the remote agent stays running. It should have been obvious, but I didn't think of it until I had a good sleep.

**Comment by hossam-magdy** [comment]
same problem encountered here too.
